### PR TITLE
added coding-standard-baseline folder for codesniffer baseline task

### PR DESCRIPTION
**Pull request by MeetKamal**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/mage-os/github-actions/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
- [ ] Yes
- [ ] No

### CODE REVIEW

File: coding-standard-baseline/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+jobs:
+  coding-standard:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: mage-os/github-actions/coding-standard-baseline@main
+          php_version: 8.1 # Optional, will be used for Php version
+          composer_version: 2
+          version: 31 # Optional, will use the latest if omitted.
+          path: app/code # Optional, will be used when event is not a pull request.
```
Comment (sprankhub) [inline coding-standard-baseline/README.md:30]: This is unused, isn't it?

File: coding-standard-baseline/action.yml [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        composer config --no-plugins allow-plugins.digitalrevolution/php-codesniffer-baseline true
+        composer require --dev "digitalrevolution/php-codesniffer-baseline: ${{ github.event.inputs.baseline_version || '*' }}"
+
+    - name: Register Coding Standard
+      shell: bash
+      run: vendor/bin/phpcs --config-set installed_paths ${{ github.workspace }}/vendor/magento/magento-coding-standard,${{ github.workspace }}/vendor/phpcompatibility/php-compatibility
+
+    - name: Get all changed files
+      id: changed-files
+      uses: tj-actions/changed-files@v35
```
Comment (sprankhub) [inline coding-standard-baseline/action.yml:83]: Could we use latest version v37 already? :)

File: coding-standard-baseline/action.yml [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        write_output_files: true
+        output_dir: ${{ github.workspace }}
+        separator: " "
+
+    - name: Verify the contents of the modified_files.txt file
+      shell: bash
+      run: |
+        sed "s/ /\n/g" ${{ github.workspace }}/modified_files.txt > ${{ github.workspace }}/newline_file.txt
+        grep -iE "\.php|\.phtml" ${{ github.workspace }}/newline_file.txt > ${{ github.workspace }}/phpcs_files.txt
+        cat ${{ github.workspace }}/phpcs_files.txt
```
Comment (sprankhub) [inline coding-standard-baseline/action.yml:94]: Should we remove this, because it is debug output?

File: coding-standard-baseline/action.yml [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+      with:
+        write_output_files: true
+        output_dir: ${{ github.workspace }}
+        separator: " "
+
+    - name: Verify the contents of the modified_files.txt file
+      shell: bash
+      run: |
+        sed "s/ /\n/g" ${{ github.workspace }}/modified_files.txt > ${{ github.workspace }}/newline_file.txt
+        grep -iE "\.php|\.phtml" ${{ github.workspace }}/newline_file.txt > ${{ github.workspace }}/phpcs_files.txt
```
Comment (sprankhub) [inline coding-standard-baseline/action.yml:93]: If you search inside https://github.com/magento/magento-coding-standard/blob/0f81833b28d6fc3d799986ebcfa7fee659927ff4/Magento2/ruleset.xml#L293 for `<include-pattern>`, you see that the coding standard checks more files than just `.php` and `.phtml`. Could we include all?

File: coding-standard-baseline/action.yml [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        cat ${{ github.workspace }}/phpcs_files.txt
+
+    - name: Checkout - Before Merge
+      shell: bash
+      id: previous-commit-files
```
Comment (sprankhub) [inline coding-standard-baseline/action.yml:98]: Since we do not use these `id`s, should we just remove them from all steps?

**Review by sprankhub — CHANGES_REQUESTED** [review]

File: coding-standard-baseline/README.md (hunk 1 above)
Comment (MeetKamal) [reply coding-standard-baseline/README.md:30]: Hi @sprankhub  -  we have added our comments in README.md file,  if not required then we can remove that.

File: coding-standard-baseline/action.yml [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+    default: "8.1"
+    description: "PHP version used to do the coding standard check."
+
+  composer_version:
+    type: string
+    required: true
+    default: "2"
+    description: "The version of composer to use."
+
+  path:
```
[the remark this answers is not in the document]
Comment (MeetKamal) [reply coding-standard-baseline/action.yml:18]: Hi @sprankhub  - for the PHP setup it requires composer_version for tools

[the remark this answers is not in the document]
Comment (MeetKamal) [reply coding-standard-baseline/action.yml:18]: tools: composer:v${{ inputs.composer_version }}

File: coding-standard-baseline/action.yml (hunk 2 above)
Comment (MeetKamal) [reply coding-standard-baseline/action.yml:83]: okay will update with latest version

File: coding-standard-baseline/action.yml (hunk 3 above)
Comment (MeetKamal) [reply coding-standard-baseline/action.yml:94]: sure will update

File: coding-standard-baseline/action.yml (hunk 5 above)
Comment (MeetKamal) [reply coding-standard-baseline/action.yml:98]: sure will remove that

File: coding-standard-baseline/action.yml (hunk 4 above)
Comment (MeetKamal) [reply coding-standard-baseline/action.yml:93]: Hi @sprankhub  - Not sure, if that recognized by php code sniffer baseline. I guess that only allowed for php code base. we need to check with Vinai on this.

**Review by MeetKamal — COMMENTED** [review]
Hi @sprankhub 

I added comments, please review.

File: coding-standard-baseline/action.yml (hunk 5 above)
Comment (MeetKamal) [reply coding-standard-baseline/action.yml]: this is done

File: coding-standard-baseline/action.yml (hunk 3 above)
Comment (MeetKamal) [reply coding-standard-baseline/action.yml]: this is done

File: coding-standard-baseline/action.yml (hunk 2 above)
Comment (MeetKamal) [reply coding-standard-baseline/action.yml]: this is done

File: coding-standard-baseline/action.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (sprankhub) [reply coding-standard-baseline/action.yml:18]: If I am not mistaken, you never use a `path` parameter anywhere. Hence, it should be safe to remove it.

File: coding-standard-baseline/README.md (hunk 1 above)
Comment (sprankhub) [reply coding-standard-baseline/README.md:30]: If I am not mistaken, you never use a `path` parameter anywhere. Hence, it should be safe to remove it.

File: coding-standard-baseline/action.yml (hunk 4 above)
Comment (sprankhub) [reply coding-standard-baseline/action.yml:92]: You can also just try to trigger one of the warnings in e.g. XML files. Other file types are definitely also checked. E.g., I already saw warnings in mail templates, which are `.html` files, myself.

**Comment by sprankhub** [comment]
Thanks, @MeetKamal. I resolved everything, which is done. Please check the remaining open comments.

**Review by MeetKamal — COMMENTED** [review]
Hi @sprankhub  - Please review my comments and changes.

**Review by sprankhub — DISMISSED** [review]

**Comment by sprankhub** [comment]
@damienwebdev, would you also have a look before we merge this?

File: coding-standard-baseline/README.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    branches:
+    - main
+
+jobs:
+  coding-standard:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: mage-os/github-actions/coding-standard-baseline@main
+          php_version: 8.1 # Optional, will be used for Php version
+          composer_version: 2
```
Comment (damienwebdev) [inline coding-standard-baseline/README.md:28]: Given that these are strings, it's probably a good idea to use the string quotes to be clear that they're not numeric. especially for things like composer.

File: coding-standard-baseline/action.yml [hunk 8]
```diff
@@ -0,0 +1,118 @@
+name: "M2 Coding Standard"
+author: "Mage-OS"
```
Comment (damienwebdev) [inline coding-standard-baseline/action.yml:2]: Can you add a test of this action against the fake mage-os package source just to prove that it works?

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: coding-standard-baseline/README.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+  pull_request:
+    branches:
+    - main
+
+jobs:
+  coding-standard:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: mage-os/github-actions/coding-standard-baseline@main
+          php_version: 8.1 # Optional, will be used for Php version
```
[the remark this answers is not in the document]
Comment (MeetKamal) [reply coding-standard-baseline/README.md:27]: Thanks @damienwebdev  - will fix and update

File: coding-standard-baseline/action.yml (hunk 8 above)
Comment (MeetKamal) [reply coding-standard-baseline/action.yml:2]: Hi @damienwebdev ,  we have tested on local and below are results 

https://github.com/MeetKamal/mageos-magento2/actions/runs/5507076138/jobs/10036558694

**Review by MeetKamal — COMMENTED** [review]
Hi @damienwebdev 

Updated changes , please review

File: coding-standard-baseline/action.yml (hunk 8 above)
Comment (sprankhub) [reply coding-standard-baseline/action.yml:2]: Let's move forward and merge this. If they do not work for whatever reason, a new PR needs to fix it.

**Review by sprankhub — APPROVED** [review]

**Pull request merged**